### PR TITLE
nr: adds syscall constants for mips (32b)

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Comment by lucab** [comment]
Correct me if wrong, but I seems to understand those constants are for mips32 ABI only. Is the "mips" arch identifier valid for both mips32 and mips64? If so, can we AND the arch width in the conditional?

Also, I'd be happy if you could run the testsuite as root and check that everything is green there.

**Comment by bkchr** [comment]
Based on the following file https://github.com/rust-lang/rust/blob/master/src/test/run-pass/conditional-compile-arch.rs, I would assume that mips is for 32bit and mips64 for 64bit.

I cross compiled the tests and executed the ambient, bounding and effective tests per hand and all worked :)

File: examples/manipulate_sys_nice.rs [hunk 1]
```diff
@@ -50,11 +50,17 @@ fn main() {
 
 fn renice(prio: libc::c_int) -> libc::c_int {
     // This is not proper logic, as it does not drain errno.
+    #[cfg(target_arch = "mips")]
+    return unsafe { libc::setpriority(libc::PRIO_PROCESS as i32, 0, prio) };
```
Comment (lucab) [inline examples/manipulate_sys_nice.rs]: Are you building this with a musl libc by chance? According to libc source this should be u32 on all Linux arches (which doesn't match my manpage), but i32 on musl. If so, the cfg conditional should be targeting that (if possible).

Comment (lucab) [inline examples/manipulate_sys_nice.rs]: And as we are repeating this in different place, I'd just define a conditional `PROC_ID` constant and just use that.

**Comment by lucab** [comment]
Great feedback! Thanks for cross-checking/testing this, and also for the the example. I just have a minor comment on that, otherwise all green.

**Comment by bkchr** [comment]
Nice find with the cfg conditional! You were right, I was compiling with musl. Hope the changes are now mergable :)

**Comment by lucab** [comment]
Yup, LGTM. I'll follow up with libc-rs to inquiry about non-musl type as I think the `u32` there may just be a bug. Thanks for you contribution!

**Comment by lucab** [comment]
While at it, do you have by chance an armv7 board nearby and willing to validate https://github.com/lucab/caps-rs/issues/2?

**Comment by bkchr** [comment]
No sorry, I don't have an armv7 board nearby.

**Pull request merged**